lib/jobs/ipmi-job.js

```diff
@@ -184,20 +184,117 @@ function ipmiJobFactory(
     };
 
     /**
-     * Collect SEL entries list from IPMI
+     * gets the SEL information using the "sel get" ipmi command
      * @memberOf IpmiJob
      *
      * @param data
-     * @param count
+     * @param stringifiedEntries
      */
-    IpmiJob.prototype.collectIpmiSel = function(data, count) {
-        count = count || 25;
-        return ipmitool.sel(data.host, data.user, data.password, count)
-        .then(function (sel) {
-            return parser.parseSelData(sel);
-        });
+    IpmiJob.prototype.getSelEntries = function(data, stringifiedEntries) {
+        return ipmitool.sel(data.host, data.user, data.password,  stringifiedEntries)
+    };
+
+    /**
+     * gets the raw information of a single sel entry
+     * @memberOf IpmiJob
+     *
+     * @param data
+     * @param recordId
+     */
+    IpmiJob.prototype.getRawSelEntry = function(data, recordId) {
+        return ipmitool.readRawSel(data.host, data.user, data.password, recordId)
+    };
+
+    /**
+     * Collect SEL entries list from IPMI and add two properties ("Sensor Type Code" and "Event Type Code") to each sel entry
+     * @memberOf IpmiJob
+     *
+     * @param data
+     */
+    IpmiJob.prototype.collectIpmiSel = function(data) {
+        var self = this;
+        var stringifiedEntries = "";
+        var lastUpdatedSelDate;
+        var currentSelDate;
+        var lastReportedSelEntryID;
+        var totalEntries;
+        var lastUpdatedSelDate;
+        var sel;
+
+        return Promise.all([
+           self.collectIpmiSelInformation(data),
```

Comment:
HI @benbp,
It doesn't get the entire SEL. It gets the range starting from the last SEL record that was read (lastReportedSelEntryID, and it is saved in the database) into the ed of the SEL.  
I tested this new implementation by reading a filled-up SEL log and it works.
I ran into an issue where it was exceeding the max buffer size of the process , but increased its size in the ipmitool file (the default 200K to 400K):
https://github.com/tannoa2/on-tasks/blob/bca88dff4b0499ea89566e070f0b2715c394cc1a/lib/utils/job-utils/ipmitool.js#L35

But Again there is will be a rare case of reading the whole SEL, we are only reading the newly reported entries as specifed by the spec :
"OnRack shall ignore all existing SEL events and only interrogate new SEL events from the time the IPMI poller starts. Each event shall only be reported on once"